src/coreclr/jit/emitriscv64.cpp

```diff
@@ -3829,12 +3816,23 @@ void emitter::emitDispInsHex(instrDesc* id, BYTE* code, size_t sz)
 void emitter::emitDispIns(
     instrDesc* id, bool isNew, bool doffs, bool asmfm, unsigned offset, BYTE* pCode, size_t sz, insGroup* ig)
 {
-    // RISCV64 implements this similar by `emitter::emitDisInsName`.
-    // For RISCV64 maybe the `emitDispIns` is over complicate.
-    // The `emitter::emitDisInsName` is focused on the most important for debugging.
-    NYI_RISCV64("RISCV64 not used the emitter::emitDispIns");
+    if (pCode == nullptr)
```

Comment:
It is good. However, how about displaying all instructions of methods in `a new function` or `caller function`? 
As I know, `emitDispIns` is too complicate for LOONGARCH64. LOONGARCH64 implemented a new function `emitDisInsName` instead of `emitDispIns`. And I just follow LOONGARCH64 for RISCV64. So RISCV64's Implementation is too different from the other archs's. Moreover, `emitDispIns` displays single instruction in the other archs.

Reply:
I agree that the `emitDispIns` function has too much utilities and ultimately can be viewed as an overkill for the RISC64's platform. On the other hand all of the other architectures (including [LoongArch64](https://github.com/dotnet/runtime/blob/6249fd2e59ed1308eb7cf976fb48c1ecab5eaa4e/src/coreclr/jit/emitloongarch64.cpp#L5958)) has implemented this function (although it seems that the LA64 doesn't use it anywhere which is strange considering the function code resemblance to some other parts of their code like [here](https://github.com/dotnet/runtime/blob/6249fd2e59ed1308eb7cf976fb48c1ecab5eaa4e/src/coreclr/jit/emitloongarch64.cpp#L3774C1-L3779C6)). In my opinion it would be more beneficial to discard some of the values passed to this function and use it in the simplified form since the new reader accustomed to other archs implementation would have much better time understanding the RISC64 implementation. I agree that `emitDispIns` display only one instruction on the other archs but it also seems that on those archs the `emitOutputInstr` emits only one instruction per call which is not the case for the RISC64's implementation. I also believe that allowing this function to exist with behavior similar to the other archs will allow to decrease the number of `ifdef`'s in the `emit.cpp` file which will result with more generalized and easier to understand code

Reply:
> I also believe that allowing this function to exist with behavior similar to the other archs will allow to decrease the number of `ifdef`'s in the `emit.cpp` file which will result with more generalized and easier to understand code

:+1: -- this should always be the goal, anything else makes maintenance and development very difficult for us. The backends should behavior as similarly as possible.

Reply:
Can you please fix this so that `emitDispIns` behaves as other platforms as so that this ifdef can be removed:

https://github.com/dotnet/runtime/blob/ea13b5bd2c17e970590068d9dfce8171b663cde1/src/coreclr/jit/emit.cpp#L4197-L4198

Reply:
Ok, done